core/src/main/scala/cats/syntax/either.scala

```diff
@@ -1,13 +1,163 @@
 package cats
 package syntax
 
-import cats.data.Xor
+import cats.data.{EitherT, Ior, Validated, ValidatedNel, Xor}
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
 
 trait EitherSyntax {
   implicit def catsSyntaxEither[A, B](eab: Either[A, B]): EitherOps[A, B] = new EitherOps(eab)
+
+  implicit def catsSyntaxEitherObject(either: Either.type ): EitherObjectOps = new EitherObjectOps(either)
 }
 
 final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
+  def foreach(f: B => Unit): Unit = eab.fold(_ => (), f)
```

Comment:
Hm there's quite a few of these around. It's certainly possible to replace them but I'm a tad hesitant - is the overhead associated with the closure significant? /cc @travisbrown